### ISSUE TITLE
Field 'Date' on extended class gets overwritten with 'PublishDate' since 2.4.3

### ISSUE THREAD

**Issue by JorisDebonnet** [silverstripe/silverstripe-blog#441]
I recently updated the dependencies on one of my sites, and my client reported that suddenly something broke! I tracked it down to the update from silverstripe/silverstripe-blog 2.4.2 to 2.4.3.

The problem is that I have created a class that extends BlogPost, and defined a 'Date' field on there. I guess that may sound confusing, but the point is that the class represents an Event, so it makes sense for the post about it to be published before the event actually takes place. Now, this works perfectly fine until 2.4.2, but since 2.4.3, whatever is filled in in "Date" gets overwritten by "PublishDate". Resulting in wrong information on the website, of course :) 

I assume that the code is using a temporary value or something which is stored in $this->Date, not thinking about the possibility that this could actually be used for something else. In any case, I would say this is an API change (even if it isn't encountered often), so should probably be fixed.

**Comment by JorisDebonnet** [comment]
Oh, I may actually be reading this wrong - it seems the issue is not when writing, but when reading. 

I expect ->Date to read my custom field, but 2.4.3 makes ->Date actually return the PublishDate.

**Comment by robbieaverill** [comment]
Possibly this commit that's causing this behaviour? https://github.com/silverstripe/silverstripe-blog/commit/299944fd1fa5f637a65f4bb8a339907a6545169e

**Comment by JorisDebonnet** [comment]
Indeed, so looks like this is related to #394 (and PR #403).

**Comment by micmania1** [comment]
Sorry @JorisDebonnet, we should have done a minor release for that. However, I'd like to avoid reverting it in case we break somebody else's code and re-introduce the RSS issue.

You should be able to fix it in your subclass with:
```
public function getDate() {
    return $this->getField('Date');
}
```

Happy to hear feedback or input from anybody else if they think a different approach would be better?

**Comment by dhensby** [comment]
This is a feature, not a bug

**Comment by dhensby** [comment]
For more information on my reasoning:

1. The RSSFeed generator needs a field called `Date` for it to render the RSS feed properly. `BlogPost` is using the method `getDate` to present a value for the `RSSFeed` - this is a feature.
2. This essentially makes `Date` a reserved field and you can't add a field to a `BlogPost` called `Date`

Unfortunately the non-core modules aren't quite as strict with semver in this case and this was released as a patch by mistake.

We *could* look at a patch where we look to see if there is a real field called `Date`, if there is, use that otherwise use `PublishedDate`

**Comment by JorisDebonnet** [comment]
I guess that for #394, two solutions were suggested, and the quickest one was implemented, which changed the API (in the sense that it added a reserved field). I guess the other solution could be implemented instead (adding a separate RSS template that overwrites the framework one), but of course some people might be counting on BlogPost having a getDate method exposed now. Well, guess I should have been there a year ago then :) 

I was able to circumvent it in my own application with @micmania1's suggestion, thanks for that. I still wonder whether something needs to be done to avoid other developers arriving at the same issue, but I guess it won't happen so often as they'll soon start with a BlogPost version that already has this (and so they'll see the issue while developing).
